msteams-platform/tabs/how-to/using-teams-client-sdk.md

```diff
@@ -1,72 +1,469 @@
+
+* Check [isSupported()](#differentiate-your-app-experience) on a capability before using it
+* Catch and handle errors when calling TeamsJS and HTML5 APIs
+
+When an API is unsupported or throws an error, add logic to fail gracefully or provide a workaround. For example:
+
+* Direct the user to your app's website
+* Direct the user to use the app in Teams to complete the flow
+* Notify the user the functionality isn't yet available
+
+Additionally, best practice is to ensure your app manifest only specifies the device permissions it's using.
+
+#### Deep linking
+
+Prior to TeamsJS version 2.0, all deep linking scenarios were handled using `shareDeepLink` (to generate a link *to* a specific part of your app) and `executeDeepLink` (to navigate to a deeplink *from* or *within* your app). TeamsJS v.2.0 introduces a new API, `navigateToApp`, for navigating to pages (and subpages) within an app in a consistent way across app hosts (Office and Outlook, in addition to Teams). Here's the updated guidance for deep linking scenarios:
+
+##### Deep links into your app
+
+Use `pages.shareDeepLink` (known as *shareDeepLink* prior to TeamsJS v.2.0) to generate and display a copyable link for the user to share. When clicked, a user will be prompted to install the app if it's not already installed for the application host (specified in the link path).
+
+##### Navigation within your app
+
+Use the new `pages.navigateToApp` API to navigate within your app within the hosting application.
+
+This API provides the equivalent of  navigating to a deep link (as the now deprecated *executeDeepLink* was once used for) without requiring your app to construct a URL or manage different deep link formats for different application hosts.
+
+##### Deep links out of your app
+
+For deep links from your app to various areas of its current host, use the strongly typed APIs provided by the TeamsJS SDK. For example, use the *Calendar* capability to open a scheduling dialog or calendar item from your app.
+
+For deep links from your app to other apps running in the same host, use `pages.navigateToApp`.
+
+For any other external deep linking scenarios, you can use `app.openLink`, which provides similar functionality to the now deprecated (starting in TeamsJS v.2.0) *executeDeepLink* API.
+
+#### Dialogs
+
+Starting with version 2.0 of TeamsJS, the Teams platform concept of [task module](../../task-modules-and-cards/what-are-task-modules.md) has been renamed to *dialog* for better consistency with existing concepts across the Microsoft 365 developer ecosystem. Accordingly, the `tasks` namespace has been deprecated in favor of the new `dialog` namespace.
+
+This new dialog capability is split into a main capability (`dialog`) for supporting HTML-based dialogs and a subcapability, `dialog.bot`, for bot-based dialog development.
+
+The dialog capability doesn't yet support adaptive card based dialogs. Adaptive-card based dialogs still need to be invoked using `tasks.startTask()`.
```

Comment:
nit: should there be a dash between "Adaptive" and "card" or between "card" and "based", i.e., "Adaptive card-based"? If so, I think it should be consistent with the previous sentence.